src/relay/pass/quantize.cc

```diff
@@ -693,18 +693,9 @@ Expr MulRealize(const Call& ref_call,
     const auto* rhs = new_args[1].as<QRealizeIntExprNode>();
     Expr ldata = lhs->data;
     Expr rdata = rhs->data;
-
     DataType dtype = cfg->dtype_activation;
-    if (lhs->dtype == Float(32)) {
-      ldata = Cast(ldata, dtype);
-    } else {
-      CHECK_EQ(lhs->dtype, dtype);
-    }
-    if (rhs->dtype == Float(32)) {
-      rdata = Cast(rdata, dtype);
-    } else {
-      CHECK_EQ(rhs->dtype, dtype);
-    }
+    ldata = Cast(ldata, dtype);
+    rdata = Cast(rdata, dtype);
```

Comment:
I discussed with @ajtulloch offline. We may have overflow issue when case fp32 directly. So maybe we should add range check before cast.

Reply:
What is the failure mode in this case (e.g., if there is overflow, is it considered fatal or do we replace with a clamp)?